tests/Microsoft.Identity.Test.Unit/PublicApiTests/PublicClientApplicationTests.cs

```diff
@@ -843,6 +843,64 @@ public async Task AcquireTokenSilent_EmptyLoginHint_TestAsync()
                 .ConfigureAwait(false);
         }
 
+        /// <summary>
+        /// Cache state:
+        ///
+        /// 2 users have acquired tokens
+        /// 1 of them is a guest in another tenant => 1 request for each tenant
+        ///
+        /// The 2 accounts have wam ids in account in the response
+        /// </summary>
+        /// <returns></returns>
+        [TestMethod]
+        [DeploymentItem(@"Resources\TokenCacheWithWamId.json")]
+        public async Task WamIdInAccountInAuthResultTestAsync()
+        {
+            const string tenant1 = "72f988bf-86f1-41af-91ab-2d7cd011db47";
+            const string tenant2 = "49f548d0-12b7-4169-a390-bb5304d24462";
+            string tenantedAuthority1 = $"https://login.microsoftonline.com/{tenant1}/";
+            string tenantedAuthority2 = $"https://login.microsoftonline.com/{tenant2}/";
+
+            using (var httpManager = new MockHttpManager())
+            {
+                // Arrange
+                PublicClientApplication pca = CreatePcaFromFileWithAuthority(httpManager, tokenCacheFile: "TokenCacheWithWamId.json");
+
+                // Act
+                var accounts = await pca.GetAccountsAsync().ConfigureAwait(false);
+                var account = accounts.Single(a => a.HomeAccountId.TenantId == tenant1);
+                var tenantProfiles = account.GetTenantProfiles();
+
+                AuthenticationResult response = await
+                    pca.AcquireTokenSilent(new[] { "User.Read" }, account)
+                    .WithAuthority(tenantedAuthority1)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                // Assert
+                Assert.AreEqual(tenant1, response.TenantId);
+                AssertWamIds(response.Account as Account, 1, "wamId1");
+                AssertTenantProfiles(tenantProfiles, tenant1, tenant2);
+                AssertTenantProfiles(response.Account.GetTenantProfiles(), tenant1, tenant2);
+                Assert.AreEqual(tenant1, response.ClaimsPrincipal.FindFirst("tid").Value);
+
+
+                // Act
+                accounts = await pca.GetAccountsAsync().ConfigureAwait(false);
+                account = accounts.Single(a => a.HomeAccountId.TenantId == tenant2);
+                response = await
+                    pca.AcquireTokenSilent(new[] { "User.Read" }, account)
+                    .WithTenantId(tenant2)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                // Assert
+                AssertWamIds(response.Account as Account, 1, "wamId2");
+                Assert.AreEqual(tenant2, response.TenantId);
```

Comment:
Assert tenant profiles for second account too?

Reply:
The second account will only have 1 tenant profile since only 1 id token is present for the home account id. That is why it is not verified for multiple tenant profiles. This is same behavior as existing tests on the cache for tenant profiles.